<center> <img src = "https://www.lupacity.com/wp-content/uploads/2015/05/iteso.jpg" align="center" width="500" height="480"/> </center>

<center><font color=#555555><font size=6> MAF1731 - Microestructura y Sistemas de Trading </font> <br> <br>
    
<center><font color=#555555><font size=5> Licenciatura en Ingeniería Financiera </font> <br> <br> <br> <br>

<center><font color=#555555><font size=6> I. F. Ricardo Enrique Cárdenas Ramos  </font> <br> <br>
   
<center> <font color= #555555> <font size = 4> Octubre 2022 </a> | <font color= #555555> <font size = 4> Repositorio: <a href = 'https://github.com/cardenasramosr/MyST_LAB_3'>Link</a></font>

<hr style="border:0.02in solid gray"> </hr>

<center> <font color= #555555> <font size = 6> Laboratorio 3. Finanzas Conductuales </font> <br> <br> <font color= #555555> <font size = 5> Desempeño en la Actividad de Trading Discrecional </font>

<hr style="border:0.02in solid gray"> </hr>

<br>

<font color= #555555> <font size = 4> Abstract </font>

<font color= #6B6B6B> <font size = 3> Se mide el desempeño de la actividad de trading realizada durante el Laboratorio 2: Trading Discrecional. Se generan estadísticas descriptivas con el archivo histórico de operaciones de cada trader. Se cuantifican ciertas métricas de atribución al desempeño de las operaciones de trading discrecional. Se identifica si los traders cayeron o no en el sesgo conductual que lleva a la mayoría de operadores a acortar las ganancias y prolongar las pérdidas. Visuales interactivos.  </font>

<br>

# <font color= #6B6B6B> <font size = 6> 1. Introducción </font>

<hr style="border:0.02in solid gray"> </hr>

Este Laboratorio tiene por objetivo el análisis exhaustivo de las operaciones realizadas por los traders Ricardo, Emilio y Leonardo durante todo el Laboratorio 2: Trading Discrecional. Para esto se elaborarán algunas funciones básicas que servirán a manera de análisis exploratorio y descriptivo inicial de los trades efectuados. Posteriormente, se entregan como resultados la evolución diaria del capital de los traders junto con algunas medidas de atribución a su desempeño. Por último, se cuantifica un sesgo cognitivo de la economía conductual que habla acerca de cómo algunos operadores tienden a acortar sus ganancias mientras mantienen abiertas sus pérdidas. 

<br>

# <font color= #6B6B6B> 2. Instalar/Cargar Paquetes y Dependencias </font>

<hr style="border:0.02in solid gray"> </hr>

<br>

## <font color= #6B6B6B> 2.1 Paquetes de Python </font> <font color= #555555>

Para correr este archivo, es necessario tener instaladas y/o tener el archivo `the_requirements.txt_` con las siguientes librerías:

- pandas
- numpy 
- yfinance
- datetime
- plotly

## <font color= #6B6B6B> 2.2 Dependencias de Archivos </font> <font color= #555555>

Las siguientes son dependencias de archivos que se necesitan para correr este cuaderno de Jupyter:

- files/bitacora_operaciones.xlsx
- instruments_pips.csv

## <font color= #6E6E6E> 2.3 Instalar Paquetes </font>

In [1]:
# Librería main
import main

<br>

# <font color= #6B6B6B> 3. Estadística Descriptiva </font>

<hr style="border:0.02in solid gray"> </hr>

Esta sección del Laboratorio titulada `Estadística Descriptiva` tiene por objetivo la entrega de: 

+ Un **DataFrame** con toda la información histórica de operaciones y la actividad de trading. Este contenedor incluye datos de las operaciones realizadas por cada uno de los traders; como la fecha de apertura y cierre de sus posiciones, los instrumentos sobre los cuales se operó, el tamaño de la posición (lotaje) en cada trade, los tipo de operación (compra o venta) y el beneficio obtenido en cada una.

+ Un **Diccionario** con las llaves *df_1_tabla* y *df_2_ranking*. La primera llave retorna un *DataFrame* con estadísticas básicas, como el número de operaciones ganadoras y perdedoras, y algunos ratios de efectividad y proporción, que miden el desempeño del trader. La segunda llave retorna un *DataFrame* que muestra el ratio de efectividad de las operaciones realizadas con cada instrumento operado. 

## <font color= #6E6E6E> 3.1 Ricardo </font>

In [2]:
# Histórico de operaciones
datos_ricardo, dic_estadisticas_ba_ricardo = main.estadistica_descriptiva("Ricardo")
datos_ricardo.head()

opentime    ticket     item  type  size  openprice        S/L  \
0 2022-09-16 23:27:17  78276910   EURGBP  sell  0.05    0.87643    0.88643   
1 2022-09-16 23:47:47  78277127  NAT.GAS  sell  0.05    7.80600    0.00000   
2 2022-09-16 23:48:45  78277128  NAT.GAS   buy  0.05    7.87000    7.37000   
3 2022-09-19 01:39:26  78281023   GBPJPY   buy  0.05  163.37600  162.87600   
4 2022-09-19 05:10:05  78283743   USDCHF  sell  0.05    0.96558    0.97058   

         T/P           closetime  closeprice  comission  swap  profit  tiempo  \
0    0.85643 2022-09-23 17:22:25     0.88645          0  0.42  -55.04   64508   
1    0.00000 2022-09-16 23:48:12     7.87000          0  0.00  -32.00      25   
2    8.87000 2022-09-22 19:58:51     7.36900          0 -5.13 -250.50   72606   
3  164.37600 2022-09-20 10:30:34   164.37700          0  0.20   34.90   31868   
4    0.95558 2022-09-22 10:30:26     0.97058          0 -1.32  -25.76   19221   

    pips  pips_acm  profit_acm  
0 -100.2    -100.2      -55.04  
1   -6.4    -106.6      -87.04  
2  -50.1    -156.7     -337.54  
3  100.1     -56.6     -302.64  
4  -50.0    -106.6     -328.40

In [3]:
datos_ricardo.tail()

opentime    ticket    item  type  size  openprice        S/L  \
16 2022-09-23 17:08:31  78521862  GBPJPY   buy  0.05  157.64300  156.64300   
17 2022-09-23 17:12:01  78521997  GBPUSD   buy  0.05    1.10120    1.09120   
18 2022-09-23 18:59:02  78526678  EURUSD   buy  0.05    0.97187    0.96187   
19 2022-09-23 19:02:08  78526781  EURGBP  sell  0.05    0.88993    0.89993   
20 2022-09-23 23:26:11  78533366  GBPUSD   buy  0.05    1.08628    1.07628   

          T/P           closetime  closeprice  comission  swap  profit  \
16  159.64300 2022-09-23 17:42:20   156.64200          0  0.00  -34.97   
17    1.12120 2022-09-23 17:50:15     1.09117          0  0.00  -50.15   
18    0.99187 2022-09-26 03:58:10     0.96187          0 -1.24  -50.00   
19    0.86993 2022-09-26 03:42:43     0.89998          0  0.33  -54.00   
20    1.10628 2022-09-26 03:42:00     1.07626          0 -0.93  -50.10   

    tiempo   pips  pips_acm  profit_acm  
16    2029 -100.1    -562.3     -376.90  
17    2294 -100.3    -662.6     -427.05  
18   32348 -100.0    -762.6     -477.05  
19   31235 -100.5    -863.1     -531.05  
20   15349 -100.2    -963.3     -581.15

In [4]:
# Estadísticas básicas: tabla
dic_estadisticas_ba_ricardo["df_1_tabla"]

Valor                              Descripcion
Medida                                                               
Ops totales        21.000000                      Operaciones totales
Ganadoras           4.000000                    Operaciones ganadoras
Ganadoras_c         3.000000          Operaciones ganadoras de compra
Ganadoras_v         1.000000           Operaciones ganadoras de venta
Perdedoras         17.000000                   Operaciones perdedoras
Perdedoras_c        9.000000         Operaciones perdedoras de compra
Perdedoras_v        8.000000          Operaciones perdedoras de venta
Mediana (Profit)  -50.000000         Mediana de profit de operaciones
Mediana (Pips)   -100.100000           Mediana de pips de operaciones
r_efectividad       0.190476  Ganadoras totales / Operaciones totales
r_proporcion        0.235294   Ganadoras totales / Perdedoras totales
r_efectividad_c     0.142857   Ganadoras compra / Operaciones totales
r_efectividad_v     0.047619    Ganadoras venta / Operaciones totales

In [5]:
# Estadísticas básicas: ranking
dic_estadisticas_ba_ricardo["df_2_ranking"]

r_efectividad  rank
symbol                     
USDJPY              1     1
GBPJPY            0.5     2
EURUSD       0.333333     3
NAT.GAS             0     4
USDCHF              0     5
EURGBP              0     6
GBPUSD              0     7
USDCAD              0     8

### Comentarios

+ Ricardo obutvo beneficios en el $19\%$ de las operaciones que realizó (ratio de efectividad).
+ Por cada operación perdedora Ricardo ganó $0.14$ operaciones (ratio de proporción).
+ Ricardo obtuvo más operaciones ganadoras de compra que de venta ($3$ de compra y $1$ de venta).
+ En $5$ instrumentos de los $8$ operados por Ricardo se obtuvo un ratio de efectividad de $0\%$. Esto significa que en tales instrumentos no se generó beneficio alguno. Únicamente la divisa *USD-JPY* tiene un ratio de efectividad del $100\%$, esto quiere decir que todas las veces que Ricardo operó con el par generó un beneficio positivo.

## <font color= #6E6E6E> 3.2 Emilio </font>

In [6]:
# Histórico de operaciones
datos_emilio, dic_estadisticas_ba_emilio = main.estadistica_descriptiva("Emilio")
datos_emilio.head()

opentime    ticket    item  type size  openprice       S/L  \
0 2022-09-20 22:18:23  58943356  EURUSD   buy  100    0.99659    0.0465   
1 2022-09-20 22:20:59  58943418  EURUSD  sell  100    0.99664    2.8965   
2 2022-09-20 22:23:08  58949300  USDCAD   buy   25    1.33659    1.3354   
3 2022-09-20 22:27:04  58960623  GBPUSD  sell   20    1.13745    1.3876   
4 2022-09-20 22:28:35  58960638  EURJPY   buy   30  143.20600  128.8540   

        T/P           closetime  closeprice  comission  swap   profit  tiempo  \
0    2.0265 2022-09-20 22:20:23     0.99654          0   0.0  -500.00     120   
1       NaN 2022-09-20 22:34:17     0.99707          0   0.0 -4300.00     798   
2    1.3360 2022-09-20 22:51:11     1.33602          0   0.0 -1066.60    1683   
3    0.9876 2022-09-20 22:48:37     1.13885          0   0.0 -2800.00    1293   
4  147.9930 2022-09-20 22:41:15   143.27400          0   0.0  1419.71     760   

   pips  pips_acm  profit_acm  
0  -0.5      -0.5     -500.00  
1  -4.3      -4.8    -4800.00  
2  -5.7     -10.5    -5866.60  
3 -14.0     -24.5    -8666.60  
4   6.8     -17.7    -7246.89

In [7]:
datos_emilio.tail()

opentime    ticket    item  type size  openprice     S/L  \
37 2022-09-26 05:07:58  59935935  GBPUSD  sell   10    1.05834  1.0590   
38 2022-09-26 17:29:32  60102836  GBPUSD  sell   20    1.08218  1.0825   
39 2022-09-26 17:30:32  60104104  GBPUSD  sell   20    1.08121  1.0820   
40 2022-09-26 17:33:21  60124360  GBPUSD  sell   20    1.08104  1.0820   
41 2022-09-26 18:10:51  60128901  GBPUSD  sell   20    1.07946  1.0840   

       T/P           closetime  closeprice  comission  swap  profit  tiempo  \
37  1.0562 2022-09-26 05:17:25     1.05903          0   0.0  -690.0     567   
38     NaN 2022-09-26 17:29:40     1.08250          0   0.0  -640.0       8   
39  1.0810 2022-09-26 17:31:27     1.08209          0   0.0 -1760.0      55   
40     NaN 2022-09-26 17:35:40     1.08203          0   0.0 -1980.0     139   
41     NaN 2022-09-26 18:14:30     1.08414          0   0.0 -9360.0     219   

    pips  pips_acm  profit_acm  
37  -6.9 -1912.246   -68262.28  
38  -3.2 -1915.446   -68902.28  
39  -8.8 -1924.246   -70662.28  
40  -9.9 -1934.146   -72642.28  
41 -46.8 -1980.946   -82002.28

In [8]:
# Estadísticas básicas: tabla
dic_estadisticas_ba_emilio["df_1_tabla"]

Valor                              Descripcion
Medida                                                               
Ops totales        42.000000                      Operaciones totales
Ganadoras          14.000000                    Operaciones ganadoras
Ganadoras_c         5.000000          Operaciones ganadoras de compra
Ganadoras_v         9.000000           Operaciones ganadoras de venta
Perdedoras         28.000000                   Operaciones perdedoras
Perdedoras_c       16.000000         Operaciones perdedoras de compra
Perdedoras_v       12.000000          Operaciones perdedoras de venta
Mediana (Profit) -665.000000         Mediana de profit de operaciones
Mediana (Pips)     -5.300000           Mediana de pips de operaciones
r_efectividad       0.333333  Ganadoras totales / Operaciones totales
r_proporcion        0.500000   Ganadoras totales / Perdedoras totales
r_efectividad_c     0.119048   Ganadoras compra / Operaciones totales
r_efectividad_v     0.214286    Ganadoras venta / Operaciones totales

In [9]:
# Estadísticas básicas: ranking
dic_estadisticas_ba_emilio["df_2_ranking"]

r_efectividad  rank
symbol                     
GOOGL.O             1     1
AMZN.O           0.75     2
EURJPY            0.5     3
TSLA.O            0.5     4
EURUSD            0.5     5
AUDCHF           0.25     6
NZDUSD              0     7
EURGBP              0     8
USDJPY              0     9
USDMXN              0    10
GBPJPY              0    11
GBPUSD              0    12
USDCAD              0    13

### Comentarios

+ Emilio obutvo beneficios en el $33\%$ de las operaciones que realizó (ratio de efectividad).
+ Por cada operación perdedora Emilio ganó $0.5$ operaciones (ratio de proporción).
+ Emilio obtuvo más operaciones ganadoras de venta que de compra ($5$ de compra y $9$ de venta).
+ En $7$ instrumentos de los $13$ operados por Emilio se obtuvo un ratio de efectividad de $0\%$. Esto significa que en tales instrumentos no se generó beneficio alguno. Únicamente la acción *GOOGL* tiene un ratio de efectividad del $100\%$, esto quiere decir que todas las veces que Emilio operó con la acción de Alphabet generó un beneficio positivo.

## <font color= #6E6E6E> 3.3 Leonardo </font>

In [10]:
# Histórico de operaciones
datos_leonardo, dic_estadisticas_ba_leonardo = main.estadistica_descriptiva("Leonardo")
datos_leonardo.head()

opentime    ticket     item  type  size  openprice     S/L  \
0 2022-09-20 19:37:38  58786460     KO.N   buy   500      59.64     NaN   
1 2022-09-20 19:52:51  58795921   TSLA.O   buy   100     312.22  309.00   
2 2022-09-20 20:55:40  58901306  GOOGL.O  sell   300     100.25     NaN   
3 2022-09-20 21:17:14  58921824   TSLA.O  sell   100     306.93     NaN   
4 2022-09-21 16:46:29  59253543   BRKb.N  sell    50     277.94  278.79   

     T/P           closetime  closeprice  comission  swap  profit  tiempo  \
0    NaN 2022-09-21 16:35:55       60.19          0 -2.38   275.0   75497   
1  314.5 2022-09-20 20:27:50      309.00          0  0.00  -322.0    2099   
2    NaN 2022-09-22 22:59:01      101.85          0 -1.68  -480.0    7401   
3    NaN 2022-09-21 17:09:10      307.82          0 -2.73   -89.0   71516   
4    NaN 2022-09-21 17:29:03      278.80          0  0.00   -43.0    2554   

    pips  pips_acm  profit_acm  
0   55.0      55.0       275.0  
1 -322.0    -267.0       -47.0  
2 -160.0    -427.0      -527.0  
3  -89.0    -516.0      -616.0  
4  -86.0    -602.0      -659.0

In [11]:
datos_leonardo.tail()

opentime    ticket    item  type  size  openprice  S/L  T/P  \
38 2022-09-26 04:52:10  59935329  GBPUSD  sell    50   1.053154  NaN  NaN   
39 2022-09-26 05:04:05  59935742  GBPUSD   buy    20   1.059735  NaN  NaN   
40 2022-09-26 05:18:31  59936318  GBPUSD  sell    20   1.059200  NaN  NaN   
41 2022-09-26 05:19:25  59936362  GBPUSD  sell    20   1.057660  NaN  NaN   
42 2022-09-26 05:21:45  59936419  GBPUSD  sell    10   1.056800  NaN  NaN   

             closetime  closeprice  comission  swap   profit  tiempo   pips  \
38 2022-09-26 05:03:40     1.06027          0   0.0 -35580.0     690 -71.16   
39 2022-09-26 05:18:07     1.05903          0   0.0  -1410.0     842  -7.05   
40 2022-09-26 05:19:22     1.05795          0   0.0   2500.0      51  12.50   
41 2022-09-26 05:21:42     1.05709          0   0.0   1140.0     137   5.70   
42 2022-09-26 05:36:56     1.05778          0   0.0   -980.0     911  -9.80   

    pips_acm  profit_acm  
38   230.458   -73918.82  
39   223.408   -75328.82  
40   235.908   -72828.82  
41   241.608   -71688.82  
42   231.808   -72668.82

In [12]:
# Estadísticas básicas: tabla
dic_estadisticas_ba_leonardo["df_1_tabla"]

Valor                              Descripcion
Medida                                                               
Ops totales        43.000000                      Operaciones totales
Ganadoras          27.000000                    Operaciones ganadoras
Ganadoras_c         9.000000          Operaciones ganadoras de compra
Ganadoras_v        18.000000           Operaciones ganadoras de venta
Perdedoras         16.000000                   Operaciones perdedoras
Perdedoras_c        6.000000         Operaciones perdedoras de compra
Perdedoras_v       10.000000          Operaciones perdedoras de venta
Mediana (Profit)  275.000000         Mediana de profit de operaciones
Mediana (Pips)      5.500000           Mediana de pips de operaciones
r_efectividad       0.627907  Ganadoras totales / Operaciones totales
r_proporcion        1.687500   Ganadoras totales / Perdedoras totales
r_efectividad_c     0.209302   Ganadoras compra / Operaciones totales
r_efectividad_v     0.418605    Ganadoras venta / Operaciones totales

In [13]:
# Estadísticas básicas: ranking
dic_estadisticas_ba_leonardo["df_2_ranking"]

r_efectividad  rank
symbol                     
KO.N                1     1
GBPUSD           0.75     2
AMZN.O       0.666667     3
EURJPY       0.666667     4
TSLA.O       0.666667     5
USDMXN       0.333333     6
AUDCHF              0     7
WMT.N               0     8
GOOGL.O             0     9
BRKb.N              0    10

### Comentarios

+ Leonardo obutvo beneficios en el $62\%$ de las operaciones que realizó (ratio de efectividad).
+ Por cada operación perdedora Leonardo ganó $1.68$ operaciones (ratio de proporción).
+ Leonardo obtuvo más operaciones ganadoras de venta que de compra ($9$ de compra y $18$ de venta).
+ En $4$ instrumentos de los $10$ operados por Leonardo se obtuvo un ratio de efectividad de $0\%$. Esto significa que en tales instrumentos no se generó beneficio alguno. Únicamente la acción *KO* tiene un ratio de efectividad del $100\%$, esto quiere decir que todas las veces que Leonardo operó con la acción de Coca Cola generó un beneficio positivo.

<br>

# <font color= #6B6B6B> 4. Métricas de Atribución al Desempeño </font>

<hr style="border:0.02in solid gray"> </hr>

Esta sección del Laboratorio titulada `Métricas de Atribución al Desempeño` tiene por objetivo la entrega de: 

+ Un **DataFrame** con la evolución diaria del capital durante el periodo en el que el trader operó.

+ Un **DataFrame** con métricas de atribución al desempeño de la actividad de trading, como el Ratio Sharpe, el DrawDown y el DrawUp. 

## <font color= #6E6E6E> 4.1 Ricardo </font>

In [14]:
# Evolución del capital
evolucionCapital_ricardo, mad_ricardo = main.metricas_atribucion_desempeño(datos_ricardo)
evolucionCapital_ricardo

profit_d  profit_acm_d
2022-09-16      -32      99968.00
2022-09-17        0      99968.00
2022-09-18        0      99968.00
2022-09-19        0      99968.00
2022-09-20     16.2      99984.20
2022-09-21   661.52     100645.72
2022-09-22  -208.28     100437.44
2022-09-23  -313.09     100124.35
2022-09-24        0     100124.35
2022-09-25        0     100124.35
2022-09-26     -205      99919.35
2022-09-27        0      99919.35
2022-09-28        0      99919.35
2022-09-29        0      99919.35
2022-09-30        0      99919.35
2022-10-01        0      99919.35
2022-10-02        0      99919.35
2022-10-03   -500.5      99418.85

In [15]:
# Métricas de atribución al desempeño
mad_ricardo

Métrica       Valor                            Descripción
0     sharpe_original    -3.59429          Sharpe Ratio Fórmula Original
1  sharpe_actualizado     4.09155          Sharpe Ratio Fórmula Ajustada
2       drawdown_capi  2022-09-21  Fecha inicial del DrawDown de Capital
3       drawdown_capi  2022-10-03    Fecha final del DrawDown de Capital
4       drawdown_capi    -1226.87     Máxima pérdida flotante registrada
5         drawup_capi  2022-09-16    Fecha inicial del DrawUp de Capital
6         drawup_capi  2022-09-21      Fecha final del DrawUp de Capital
7         drawup_capi      677.72    Máxima ganancia flotante registrada

### Comentarios

+ Al final del periodo Ricardo terminó con una pérdida de $581.15$ dólares.
+ El Ratio de Sharpe original demuestra que las operaciones de Ricardo pierden dinero por cada unidad de riesgo asumida, lo que va en línea con el hecho de que el capital inicial de la cuenta disminuye al final de la operación.
+ La máxima pérdida flotante registrada fue por la cantidad de $1,226$ dólares.
+ La máxima ganancia flotante registrada fue por la cantidad de $677.72$ dólares.

## <font color= #6E6E6E> 4.2 Emilio </font>

In [16]:
# Evolución del capital
evolucionCapital_emilio, mad_emilio = main.metricas_atribucion_desempeño(datos_emilio)
evolucionCapital_emilio

profit_d  profit_acm_d
2022-09-20 -6246.89      93753.11
2022-09-21  1580.81      95333.92
2022-09-22 -1207.12      94126.80
2022-09-23   -26113      68013.75
2022-09-24        0      68013.75
2022-09-25        0      68013.75
2022-09-26   -50016      17997.72

In [17]:
# Métricas de atribución al desempeño
mad_emilio

Métrica       Valor                            Descripción
0     sharpe_original    -8.20108          Sharpe Ratio Fórmula Original
1  sharpe_actualizado    -6.57346          Sharpe Ratio Fórmula Ajustada
2       drawdown_capi  2022-09-21  Fecha inicial del DrawDown de Capital
3       drawdown_capi  2022-09-26    Fecha final del DrawDown de Capital
4       drawdown_capi    -77336.2     Máxima pérdida flotante registrada
5         drawup_capi  2022-09-20    Fecha inicial del DrawUp de Capital
6         drawup_capi  2022-09-21      Fecha final del DrawUp de Capital
7         drawup_capi     1580.81    Máxima ganancia flotante registrada

### Comentarios

+ Al final del periodo Emilio terminó con una pérdida de $82,002.28$ dólares.
+ El Ratio de Sharpe original demuestra que las operaciones de Emilio pierden dinero por cada unidad de riesgo asumida, lo que va en línea con el hecho de que el capital inicial de la cuenta disminuye significativamente al final de la operación.
+ La máxima pérdida flotante registrada fue por la cantidad de $77,336.2$ dólares.
+ La máxima ganancia flotante registrada fue por la cantidad de $1,580.81$ dólares.

## <font color= #6E6E6E> 4.2 Leonardo </font>

In [18]:
# Evolución del capital
evolucionCapital_leonardo, mad_leonardo = main.metricas_atribucion_desempeño(datos_leonardo)
evolucionCapital_leonardo

profit_d  profit_acm_d
2022-09-20     -322      99678.00
2022-09-21    270.5      99948.50
2022-09-22     -480      99468.50
2022-09-23 -1360.07      98108.43
2022-09-24        0      98108.43
2022-09-25        0      98108.43
2022-09-26 -70777.2      27331.18

In [19]:
# Métricas de atribución al desempeño
mad_leonardo

Métrica       Valor                            Descripción
0     sharpe_original     -6.5832          Sharpe Ratio Fórmula Original
1  sharpe_actualizado    -5.02973          Sharpe Ratio Fórmula Ajustada
2       drawdown_capi  2022-09-21  Fecha inicial del DrawDown de Capital
3       drawdown_capi  2022-09-26    Fecha final del DrawDown de Capital
4       drawdown_capi    -72617.3     Máxima pérdida flotante registrada
5         drawup_capi  2022-09-20    Fecha inicial del DrawUp de Capital
6         drawup_capi  2022-09-21      Fecha final del DrawUp de Capital
7         drawup_capi       270.5    Máxima ganancia flotante registrada

### Comentarios

+ Al final del periodo Leonardo terminó con una pérdida de $-72,668.82$ dólares.
+ El Ratio de Sharpe original demuestra que las operaciones de Leonardo pierden dinero por cada unidad de riesgo asumida, lo que va en línea con el hecho de que el capital inicial de la cuenta disminuye significativamente al final de la operación.
+ La máxima pérdida flotante registrada fue por la cantidad de $72,617.3$ dólares.
+ La máxima ganancia flotante registrada fue por la cantidad de $270.5$ dólares.

<br>

# <font color= #6B6B6B> 5. Finanzas Conductuales </font>

<hr style="border:0.02in solid gray"> </hr>

Esta sección del Laboratorio titulada `Finanzas Conductuales` tiene por objetivo la entrega de: 

+ Un **Diccionario** con las ocurrencias (en caso de existir) del sesgo propuesto por los psicólogos Kahneman y Tversky, según el cual los traders tienden a acortar sus ganancias y prolongar sus pérdidas.


## <font color= #6E6E6E> 5.1 Ricardo </font>

In [20]:
# Identificación de sesgo conductual
dic_ricardo = main.finanzas_conductuales(datos_ricardo, evolucionCapital_ricardo)
dic_ricardo

{'ocurrencias': {'cantidad': 5,
  'ocurrencia_1': {'timestamp': Timestamp('2022-09-20 10:30:34'),
   'operaciones': {'ganadora': {'instrumento': 'GBPJPY',
     'volumen': 0.05,
     'sentido': 'buy',
     'profit_ganadora': 34.9},
    'perdedora': {'instrumento': 'EURUSD',
     'volumen': 1.0,
     'sentido': 'sell',
     'profit_perdedora': -18.700000000000383}},
   'ratio_cp_profit_acm': 0.06178958498546254,
   'ratio_cg_profit_acm': -0.11531853026698388,
   'ratio_cp_cg': -0.5358166189111858},
  'ocurrencia_2': {'timestamp': Timestamp('2022-09-21 09:32:03'),
   'operaciones': {'ganadora': {'instrumento': 'EURUSD',
     'volumen': 1.0,
     'sentido': 'sell',
     'profit_ganadora': 1000.0},
    'perdedora': {'instrumento': 'USDCAD',
     'volumen': 0.5,
     'sentido': 'sell',
     'profit_perdedora': -7.099999999999884}},
   'ratio_cp_profit_acm': -0.010874559656915122,
   'ratio_cg_profit_acm': 1.5316281206922957,
   'ratio_cp_cg': -0.007099999999999884},
  'ocurrencia_3': {'times

## <font color= #6E6E6E> 5.2 Emilio </font>

In [21]:
# Identificación de sesgo conductual
dic_emilio = main.finanzas_conductuales(datos_emilio, evolucionCapital_emilio)
dic_emilio

{'ocurrencias': {'cantidad': 6,
  'ocurrencia_1': {'timestamp': Timestamp('2022-09-20 22:41:15'),
   'operaciones': {'ganadora': {'instrumento': 'EURJPY',
     'volumen': '30',
     'sentido': 'buy',
     'profit_ganadora': 1419.71},
    'perdedora': {'instrumento': 'EURUSD',
     'volumen': '100',
     'sentido': 'buy',
     'profit_perdedora': -99.99999999998899}},
   'ratio_cp_profit_acm': 0.013799022753207097,
   'ratio_cg_profit_acm': -0.19590610592957805,
   'ratio_cp_cg': -0.07043692021609271},
  'ocurrencia_2': {'timestamp': Timestamp('2022-09-20 22:41:15'),
   'operaciones': {'ganadora': {'instrumento': 'EURJPY',
     'volumen': '30',
     'sentido': 'buy',
     'profit_ganadora': 1419.71},
    'perdedora': {'instrumento': 'GBPUSD',
     'volumen': '20',
     'sentido': 'sell',
     'profit_perdedora': -131.99999999997658}},
   'ratio_cp_profit_acm': 0.018214710034232143,
   'ratio_cg_profit_acm': -0.19590610592957805,
   'ratio_cp_cg': -0.09297673468523612},
  'ocurrencia_3':

## <font color= #6E6E6E> 5.3 Leonardo </font>

In [22]:
dic_leonardo = main.finanzas_conductuales(datos_leonardo, evolucionCapital_leonardo)
dic_leonardo

{'ocurrencias': {'cantidad': 0}}

# <font color= #6B6B6B> <font size = 6> 7. Conclusiones </font>

<hr style="border:0.02in solid gray"> </hr>

<br>

# <font color= #6B6B6B> <font size = 6> 8. Referencias </font>

<hr style="border:0.02in solid gray"> </hr>

+ Instrucciones del docente.

<br>